integrations/docs/API/swagger.yaml

```diff
@@ -0,0 +1,287 @@
+tags:
+  - name: admins
+    description: Secured Admin-only calls
+  - name: developers
+    description: Operations available to regular developers
+paths:
+  /catalog:
+    get:
+      tags:
+        - developers
+      summary: schema list of loaded catalogs
+      operationId: searchCatalog
+      description: |
+        By passing in the appropriate search attributes, you can search for
+        available catalogs in the system
+      parameters:
+        - in: query
+          name: searchString
+          description: pass an optional search string for looking up catalog components
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: limit
+          description: maximum number of records to return
+          schema:
+            type: integer
+            format: int32
+            minimum: 0
+            maximum: 50
+      responses:
+        '200':
+          description: search results matching criteria
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '../../../schema/system/catalog.schema'
+        '400':
+          description: bad input parameter
+    post:
+      tags:
+        - admins
+      summary: adds a catalog item
+      operationId: addCatalog
+      description: |
+        Adds a catalog item to the system, expecting the internal URLs to be accessible
+      requestBody:
+        description: Inventory item to add
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '../../../schema/system/catalog.schema'
+      responses:
+        '201':
+          description: item created
+        '400':
+          description: 'invalid input, object invalid'
+        '409':
+          description: an existing item already exists
+  /repository:
+    get:
+      tags:
+        - developers
+      summary: repository of available integrations
+      operationId: searchRepository
+      description: |
+        By passing in the appropriate search attributes, you can search for
+        available integration in the repository (available for loading into the store)
+      parameters:
+        - in: query
+          name: searchString
+          description: pass an optional search string for looking up available Integrations
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: limit
+          description: maximum number of records to return
+          schema:
+            type: integer
+            format: int32
+            minimum: 0
+            maximum: 50
+      responses:
+        '200':
+          description: search results matching criteria
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '../../../schema/system/integration.schema'
```

Comment:
The current figma suggests having a "status" section, which indicates if this integration was added or not. Would it be trivial to add another field on this schema about whether it was added or not?

Reply:
The UI could also do this by cross-checking with the "added" list, but if it is trivial to add it to the schema/API can we do it please?